packages/@react-aria/actiongroup/src/useActionGroupItem.ts

```diff
@@ -36,6 +36,19 @@ export function useActionGroupItem<T>(props: ActionGroupItemProps, state: ListSt
     role: BUTTON_ROLES[selectionMode]
   };
 
+  // Cause rerender when focus key changes so tab index properly updates
+  let [, setFocusedKey] = useState(false);
+  useEffect(() => {
+    let handler = (focusedKey) => {
+      setFocusedKey(focusedKey);
```

Comment:
this will still cause a rerender for every time the focusedKey changes
I think we want to setFocusedKey to be setAmIFocused essentially (great name, please use something better)
and we know if the item is focused if the handler focusedKey === item key